apps/confidential/confidential_test.go

```diff
@@ -812,6 +964,105 @@ func TestWithTenantID(t *testing.T) {
 	}
 }
 
+func TestWithTenantIDNoInstanceDiscovery(t *testing.T) {
```

Comment:
TenantID creates a new authInfo in the authparam, which previously reset instanceDiscovery flag. I think it's worth keeping this test.

Reply:
That's a bug in the `authority.Info` constructor surfacing in `AuthParams.WithTenant()`. It's unrelated to combining these options because the copy returned by `WithTenant()` gets used whether the options are set or not. It would be better to catch this bug in a test of either `WithTenant` or the constructor ([here](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/4abfea36f99928ad52523e017b804616993470f5/apps/internal/oauth/ops/authority/authority_test.go#L322) or [here](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/dev/apps/internal/oauth/ops/authority/authority_test.go#L301)). I lean toward the former because `WithTenant` is responsible for returning a complete copy whereas the constructor doesn't have, and isn't necessarily required to have, a parameter for each of the struct's fields, though perhaps it should (`authority.Info` is awkward about this; problem for a future PR).

Reply:
That makes sense, I can remove the withTenantIdNoInstance test. I think inherently the "bug" is only showing up because I added the InstanceDiscovery field to the authInfo struct. In that regard, having it defaulted to false wouldn't affect anything else since that flag is only important to cases where instanceDiscovery is disabled.